dashboard/test/helpers/script_levels_helper_test.rb

```diff
@@ -17,4 +22,32 @@ class ScriptLevelsHelperTest < ActionView::TestCase
     assert_equal '//test.code.org/api/hour/finish/frozen', hoc_finish_url(Script.find_by_name('frozen'))
     assert_equal '//test.code.org/api/hour/finish/course4', hoc_finish_url(Script.find_by_name('course4'))
   end
+
+  test 'script name instead of stage name in header for HOC' do
+    self.stubs(:current_user).returns(nil)
+    @script = Script.find_by_name(Script::HOC_NAME)
+    @script_level = @script.get_script_level_by_chapter 1
+    @stage = @script_level.stage
+    @game = @script_level.level.game
```

Comment:
yeah I think that's a good idea. One nice thing about abstracting these out like you are is that we can see when the interface is hairier than we need it to be.

I think you could also make header_progress take params instead of using the instance variables, which is what you'll want to do when you make it an ajax call anyway